node/src/reactor.rs

```diff
@@ -925,54 +925,13 @@ where
         + From<PeerBehaviorAnnouncement>,
 {
     match message {
-        NetResponse::Deploy(ref serialized_item) => {
-            // Incoming Deploys should be routed to the `DeployAcceptor` rather than directly to the
-            // `DeployFetcher`.
-            let event =
-                match bincode::deserialize::<FetchResponse<Deploy, DeployId>>(serialized_item) {
-                    Ok(FetchResponse::Fetched(deploy)) => {
-                        <R as Reactor>::Event::from(deploy_acceptor::Event::Accept {
-                            deploy: Box::new(deploy),
-                            source: Source::Peer(sender),
-                            maybe_responder: None,
-                        })
-                    }
-                    Ok(FetchResponse::NotFound(deploy_id)) => {
-                        info!(%sender, ?deploy_id, "peer did not have deploy",);
-                        <R as Reactor>::Event::from(fetcher::Event::<Deploy>::AbsentRemotely {
-                            id: deploy_id,
-                            peer: sender,
-                        })
-                    }
-                    Ok(FetchResponse::NotProvided(deploy_id)) => {
-                        warn!(
-                            %sender,
-                            %deploy_id,
-                            "peer refused to provide deploy"
-                        );
-                        return effect_builder
-                            .announce_block_peer_with_justification(
-                                sender,
-                                BlocklistJustification::PeerDidNotProvideADeploy { deploy_id },
-                            )
-                            .ignore();
-                    }
-                    Err(error) => {
-                        warn!(
-                            %sender,
-                            %error,
-                            "received a deploy item we couldn't parse",
-                        );
-                        return effect_builder
-                            .announce_block_peer_with_justification(
-                                sender,
-                                BlocklistJustification::SentBadDeploy { error },
-                            )
-                            .ignore();
-                    }
-                };
-            <R as Reactor>::dispatch_event(reactor, effect_builder, rng, event)
-        }
+        NetResponse::Deploy(ref serialized_item) => handle_fetch_response::<R, Deploy>(
```

Comment:
Since the deploy doesn't go through the `DeployAcceptor` anymore, I don't think it gets validated. Does it need to get validated at this point?

Reply:
The `Deploy` gets minimal validation done by the fetcher (see `impl FetcherItem for Deploy`) when it arrives, so in this case where we've explicitly asked for this Deploy, we don't need (or want) to run the more comprehensive validation done by the `DeployAcceptor`.

Reply:
And also should we maybe get rid of the `Source::Peer(_)` machinery in `DeployAcceptor::handle_put_to_storage()` in this case?

Reply:
Well, that's the appropriate source variant when we get a deploy gossiped to us - that flow still goes via the deploy acceptor.

Reply:
I think we can look at using `Source::PeerGossiped` in that case, but certainly not the scope of this PR.